src/main/java/org/ice4j/ice/Agent.java

```diff
@@ -2441,80 +2547,14 @@ public void setGeneration(int generation)
         this.generation = generation;
     }
 
-    /**
-     * Schedule STUN checks for selected pair.
-     */
-    private void runInStunKeepAliveThread()
-    {
-        long consentFreshnessInterval = Long.getLong(
-                StackProperties.CONSENT_FRESHNESS_INTERVAL,
-                DEFAULT_CONSENT_FRESHNESS_INTERVAL);
-
-        int originalConsentFreshnessWaitInterval = Integer.getInteger(
-                StackProperties.CONSENT_FRESHNESS_ORIGINAL_WAIT_INTERVAL,
-                DEFAULT_CONSENT_FRESHNESS_ORIGINAL_WAIT_INTERVAL);
-
-        int maxConsentFreshnessWaitInterval = Integer.getInteger(
-                StackProperties.CONSENT_FRESHNESS_MAX_WAIT_INTERVAL,
-                DEFAULT_CONSENT_FRESHNESS_MAX_WAIT_INTERVAL);
-
-        int consentFreshnessMaxRetransmissions = Integer.getInteger(
-                StackProperties.CONSENT_FRESHNESS_MAX_RETRANSMISSIONS,
-                DEFAULT_CONSENT_FRESHNESS_MAX_RETRANSMISSIONS);
-
-        while (runInStunKeepAliveThreadCondition())
-        {
-            for (IceMediaStream stream : getStreams())
-            {
-                for (Component component : stream.getComponents())
-                {
-                    for (CandidatePair pair : component.getKeepAlivePairs())
-                    {
-
-                        if (pair != null)
-                        {
-                            if (performConsentFreshness)
-                            {
-                                connCheckClient.startCheckForPair(
-                                    pair,
-                                    originalConsentFreshnessWaitInterval,
-                                    maxConsentFreshnessWaitInterval,
-                                    consentFreshnessMaxRetransmissions);
-                            }
-                            else
-                            {
-                                connCheckClient
-                                    .sendBindingIndicationForPair(pair);
-                            }
-                        }
-                    }
-                }
-            }
-
-            if (!runInStunKeepAliveThreadCondition())
-            {
-                break;
-            }
-
-            try
-            {
-                Thread.sleep(consentFreshnessInterval);
-                Thread.yield();
-            }
-            catch(InterruptedException e)
-            {
-            }
-        }
-        logger.info(Thread.currentThread().getName() + " ends.");
-    }
 
     /**
-     * Determines whether {@link #runInStunKeepAliveThread()} is to run.
+     * Determines whether {@link #stunKeepAliveRunnable} is to run.
      *
-     * @return <tt>true</tt> if <tt>runInStunKeepAliveThread()</tt> is to run;
+     * @return <tt>true</tt> if <tt>{@link #stunKeepAliveRunnable}</tt> is to run;
      * otherwise, <tt>false</tt>
      */
-    private boolean runInStunKeepAliveThreadCondition()
+    private boolean runStunKeepAliveCondition()
```

Comment:
if we're gonna rename this, i'd prefer something more like `shouldRunStunKeepAlive`